src/spikeinterface/curation/curation_tools.py

```diff
@@ -59,7 +59,7 @@ def _find_duplicated_spikes_random(spike_train: np.ndarray, censored_period: int
 
 if HAVE_NUMBA:
 
-    @numba.jit((numba.int64[::1], numba.int32), nopython=True, nogil=True, cache=True)
```

Comment:
I have no idea why this is failing:

![image](https://github.com/SpikeInterface/spikeinterface/assets/6429509/90211fd5-2192-45ba-8d59-d6ae28e51534)

But also last time I checked this did not make any difference. Maybe @DradeAW wants to fix this as this was important to him.

Feel free to push this or we can do this in another PR.

Reply:
The `nopython=True, nogil=True` is important.
The other stuff is just there to reassure me ^^. I'm fine with removing them

Reply:
@alejoe91 this should be ready to merge.